your-rents-services-geodata/src/main/java/com/yourrents/services/geodata/controller/CountryController.java

```diff
@@ -0,0 +1,31 @@
+package com.yourrents.services.geodata.controller;
+
+import com.yourrents.services.geodata.exception.DataNotFoundException;
+import com.yourrents.services.geodata.model.Country;
+import com.yourrents.services.geodata.repository.CountryRepository;
+import java.util.UUID;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("${yrs-geodata.api.basepath}/countries")
+class CountryController {
+
+    private final CountryRepository countryRepository;
+
+	CountryController(CountryRepository countryRepository) {
```

Comment:
...still I don't understant this obsession of yours for having injection by constructor instead of autowired properties, but ok, I don't think we need a single style of injection.

Reply:
I do prefer injection by constructor vs by setters. 
However this specific injection was already used for injection in _CityRepository_.

Reply:
I don't have setter injection...just field injection...